api/src/test/java/org/pmiops/workbench/workspaces/WorkspaceServiceTest.java

```diff
@@ -415,4 +424,74 @@ public void deleteWorkspace_failedProjectDeletion() {
 
     verify(mockBillingProjectAuditor, never()).fireDeleteAction(eq(billingProject));
   }
+
+  @Test
+  public void updateBillingAccount_freeTierToUserOwned() throws Exception {
+    workbenchConfig.featureFlags.enableBillingUpgrade = true;
+    String newBillingAccount = "billing-123";
+    DbWorkspace workspace = dbWorkspaces.get(1); // arbitrary choice of those defined for testing
+    workspace.setBillingAccountName(workbenchConfig.billing.freeTierBillingAccountName());
+    BillingAccount billingAccount = new BillingAccount();
+    billingAccount.setOpen(true);
+    when(mockCloudBillingClient.getBillingAccount(newBillingAccount)).thenReturn(billingAccount);
+    assertThat(workspace.getBillingAccountName())
+        .isEqualTo(workbenchConfig.billing.freeTierBillingAccountName());
+
+    workspaceService.updateWorkspaceBillingAccount(workspace, newBillingAccount);
+
+    verify(mockFireCloudService)
+        .updateBillingAccount(workspace.getWorkspaceNamespace(), newBillingAccount);
+    verify(mockFireCloudService, never()).updateBillingAccountAsService(anyString(), anyString());
+    assertThat(workspace.getBillingAccountName()).isEqualTo(newBillingAccount);
+  }
+
+  @Test
+  public void updateBillingAccount_userOwnedToFreeTier() throws Exception {
+    workbenchConfig.featureFlags.enableBillingUpgrade = true;
+    String newBillingAccount = "billing-123";
```

Comment:
nit: technically "old" billing account for this test

Reply:
hahha